### ISSUE TITLE
False positive (new in 0.40.0) Declarations and declarations with annotations should have an empty space between.

### ISSUE THREAD

**Issue by NickButcher1** [pinterest/ktlint#995]
## Expected Behavior
ktlint should not report an error on this code.  Specifically, it should be valid to have no blank line between a comment and a Suppress annotation inside a function.

## Observed Behavior
ktlint reports "Declarations and declarations with annotations should have an empty space between."

## Steps to Reproduce
Run ktlint 0.39.0 on this code - no error reported.
Run ktlint 0.40.0 on this code - "Declarations and declarations with annotations should have an empty space between." reported against the annotation line.

```
// Override the fragment for...
@Suppress("ValidFragment") // Doesn't matter for unit test.
val myFragment = object : MyFragment() {
      // Some code.
}
```

## Your Environment
ktlint 0.40.0 
AGP 4.1.1
Gradle 6.7.1
Windows 10.

**Comment by NickButcher1** [comment]
I realised my mistake here (although I'm surprised that this causes these false positives).

I have the experimental flag enabled AND I have `disabled_rules=experimental:annotation` in my .editorconfig.  If I remove the disabled_rules entry from .editorconfig, the problem disappears.

**Comment by NickButcher1** [comment]
I should have said, I have the experiemant flag enabled.  If I turn that off, the problem disappears.

**Comment by shashachu** [comment]
@NickButcher1 if you have a chance, can you confirm it's the comment causing the issue? i.e. if you have
```
// Override the fragment for...
@Suppress("ValidFragment")
val myFragment = object : MyFragment() {
      // Some code.
}
```
are there any errors reported?

**Comment by NickButcher1** [comment]
This code reports the error.

```
        // Override the fragment for...
        @Suppress("ValidFragment")
        val myFragment = object : MyFragment() {
              // Some code.
        }
```

This code does not report the error.
```
        @Suppress("ValidFragment")
        val myFragment = object : MyFragment() {
              // Some code.
        }
```
This code does not report the error.
```
        @Suppress("ValidFragment") // Doesn't matter for unit test.
        val myFragment = object : MyFragment() {
              // Some code.
        }
```
So it appears that the comment on the line before the Suppress annotation is the problem.

**Comment by shashachu** [comment]
@NickButcher1 I'm unable to repro this with 0.40.0 and also I realized the rule was checked in with a unit test covering this case: https://github.com/pinterest/ktlint/blob/master/ktlint-ruleset-experimental/src/test/kotlin/com/pinterest/ktlint/ruleset/experimental/SpacingBetweenDeclarationsWithAnnotationsRuleTest.kt#L37

Is it possible it requires some additional context in order to trigger the bug?

**Comment by shashachu** [comment]
Can't repro on lastest.
